src/ReflectionClosure.php

```diff
@@ -84,27 +84,30 @@ public function getCode()
         }
 
         $className = null;
-        $fn = false;
-
 
         if (null !== $className = $this->getClosureScopeClass()) {
             $className = '\\' . trim($className->getName(), '\\');
         }
 
+        $fn = false;
+        $modern_php = false;
 
-        if($php7 = PHP_MAJOR_VERSION === 7){
-            switch (PHP_MINOR_VERSION){
-                case 0:
-                    $php7_types = array('string', 'int', 'bool', 'float');
-                    break;
-                case 1:
-                    $php7_types = array('string', 'int', 'bool', 'float', 'void');
-                    break;
-                case 2:
-                default:
-                    $php7_types = array('string', 'int', 'bool', 'float', 'void', 'object');
+        if (version_compare(PHP_VERSION, '7.0.0', '>=')) {
+            $modern_php = true;
+
+            if (version_compare(PHP_VERSION, '7.4', '>=')) {
+                $fn = true;
+            }
+
+            if (version_compare(PHP_VERSION, '7.1', '<')) {
+                $scalar_types = array('string', 'int', 'bool', 'float');
+            } elseif(version_compare(PHP_VERSION, '7.2', '<')) {
+                $scalar_types = array('string', 'int', 'bool', 'float', 'void');
+            } elseif (version_compare(PHP_VERSION, '7.4', '<=')) {
+                $scalar_types = array('string', 'int', 'bool', 'float', 'void', 'object');
+            } else {
+                $scalar_types = array('string', 'int', 'bool', 'float', 'void', 'object', 'mixed');
```

Comment:
not all types from here are scalars, the var name should probably be `$php_types`

Reply:
Just call them `builtinTypes`. That's what PHP calls them.